src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -387,5 +387,15 @@ public static bool IsCollectionType(CSharpType type)
         {
             return type.IsFrameworkType && (IsDictionary(type) || IsList(type));
         }
+
+        public static bool RequiresToList(CSharpType from, CSharpType to)
+        {
+            if (!to.IsFrameworkType || !from.IsFrameworkType || from.FrameworkType != typeof(IEnumerable<>))
+            {
+                return false;
+            }
+
+            return to.FrameworkType == typeof(IReadOnlyList<>) || to.FrameworkType == typeof(IList<>);
```

Comment:
No, `IsList` won't work because it returns `true` for `IEnumerable<T>`. In this case, if both `from` and `to` are `IEnumerable<T>`, we don't need conversion.